custom/icds_reports/ucr/expressions.py

```diff
@@ -288,10 +291,37 @@ class GetAppVersion(JsonObject):
     def configure(self, app_version_string):
         self._app_version_string = app_version_string
 
-    def __call__(self, item, context=None):
+    def get_version_from_app_object(self, item):
+        domain_name = item.get('domain')
+        form_accessor = FormAccessors(domain_name)
+        form_id = item.get('form').get('meta').get('instanceID')
+        form = form_accessor.get_form(form_id)
+
+        app = get_app(domain_name, form.app_id)
+
+        if is_linked_app(app):
+            app_version = app.upstream_version
+        else:
+            app_version = app.version
+
+        return app_version
+
+    def get_version_from_appversion_string(self, item, context):
         app_version_string = self._app_version_string(item, context)
         return get_version_from_appversion_text(app_version_string)
 
+    def __call__(self, item, context=None):
+        if settings.SERVER_ENVIRONMENT == 'india':
+            try:
+                app_version = self.get_version_from_app_object(item)
+            except Http404:
```

Comment:
When does a 404 happen?

Reply:
explained here: https://github.com/dimagi/commcare-hq/pull/26889#discussion_r394399372

Reply:
but 404 means the app wasn't found? was mostly just confused because there is not a web request so curious what function was raising that

Reply:
Ow yeah.. it was get_app function.

Reply:
It's a kind of weird quirk of `get_app`. There are app manager exceptions for `ModuleNotFound` and `FormNotFound` but not for when an application isn't found.